code/modules/food_and_drink/snacks.dm

```diff
@@ -1298,13 +1298,54 @@ ABSTRACT_TYPE(/obj/item/reagent_containers/food/snacks/soup)
 	heal_amt = 1
 	food_effects = list("food_explosion_resist")
 	meal_time_flags = MEAL_TIME_BREAKFAST
+	var/random_bagel = TRUE
 
 	New()
 		..()
-		if(rand(1,3) == 1)
-			src.icon_state = "seedbagel"
-			src.name = "seed bagel"
-			src.desc = "A bagel. But with seeds on it!"
+		if (random_bagel)
+			if(prob(33))
+				src.icon_state = "seedbagel"
+				src.name = "seed bagel"
+				src.desc = "A bagel. But with seeds on it!"
+
+	attackby(obj/item/W, mob/user)
+		if(istype(W,/obj/item/reagent_containers/food/snacks/condiment/cream))
```

Comment:
Should check if there is sufficient cream to give a good schmear?

Reply:
This recipe just uses the Cream condiment item, which is made of 10u milk reagent in the processor and contains 10u cream reagent. I just added the 10u cream reagent to the cream cheese bagel when it's created.

Reply:
(And I just checked, you can't remove cream reagent from Cream the object. It must always be sufficiently creamy.)